src/parsimmon.js

```diff
@@ -986,6 +986,16 @@ _.chain = function(f) {
   });
 };
 
+_.assert = function(f, err) {
+  assertFunction(f);
+  if (err !== undefined) {
+    assertString(err);
+  }
+  return this.chain(function(value) {
+    return f(value) ? Parsimmon.succeed(value) : Parsimmon.fail(err);
+  });
+};
+
```

Comment:
I think we should `assertFunction(f)` and make `err` a required parameter. Otherwise it's not going to have a useful error message when assert fails.